Makefile

```diff
@@ -0,0 +1,112 @@
+# Run `make help` to display help
+
+# --- Global -------------------------------------------------------------------
+O = out
+COVERAGE = 0
+VERSION ?= $(shell git describe --tags --dirty  --always)
+REPO_ROOT = $(shell git rev-parse --show-toplevel)
+
+all: build test check-coverage lint  ## build, test, check coverage and lint
+	@if [ -e .git/rebase-merge ]; then git --no-pager log -1 --pretty='%h %s'; fi
+	@echo '$(COLOUR_GREEN)Success$(COLOUR_NORMAL)'
+
+ci: clean check-uptodate all  ## Full clean build and up-to-date checks as run on CI
+
+clean::  ## Remove generated files
+	-rm -rf $(O)
+
+.PHONY: all ci clean
+
+# --- Build --------------------------------------------------------------------
+GO_LDFLAGS = -X main.version=$(VERSION)
+CMDS = .
+
+build: | $(O)  ## Build reflect binaries
+	go build -o $(O) -ldflags='$(GO_LDFLAGS)' $(CMDS)
+
+install:  ## Build and install binaries in $GOBIN
+	go install -ldflags='$(GO_LDFLAGS)' $(CMDS)
+
+.PHONY: build install
+
+# --- Test ---------------------------------------------------------------------
+COVERFILE = $(O)/coverage.txt
+
+test: $(O) ## Run tests and generate a coverage file
+	go test -coverprofile=$(COVERFILE) ./...
+
+check-coverage: test  ## Check that test coverage meets the required level
+	@go tool cover -func=$(COVERFILE) | $(CHECK_COVERAGE) || $(FAIL_COVERAGE)
+
+cover: test  ## Show test coverage in your browser
+	go tool cover -html=$(COVERFILE)
+
+check-uptodate: protos
+	test -z "$$(git status --porcelain)"
+
+CHECK_COVERAGE = awk -F '[ \t%]+' '/^total:/ {print; if ($$3 < $(COVERAGE)) exit 1}'
+FAIL_COVERAGE = { echo '$(COLOUR_RED)FAIL - Coverage below $(COVERAGE)%$(COLOUR_NORMAL)'; exit 1; }
+
+.PHONY: check-coverage check-uptodate cover test
+
+# --- Lint ---------------------------------------------------------------------
+
+lint:  ## Lint go source code
+	golangci-lint run
+
+.PHONY: lint
+
+# --- Protos ---------------------------------------------------------------------
+
+lint-protos:  ## Lint *.proto files
+	buf lint proto
+
+protos: lint-protos ## Generate Go pb and grpc binding for .proto files
+	protoc \
+		-I proto \
+		--go_out=paths=source_relative:pb \
+		--go-grpc_out=paths=source_relative:pb \
+		$(shell find proto -name google -prune -o -regex '.*/[^_].*\.proto' -print)
+	gofumpt -w .
+
+.PHONY: proto
+
+# --- Release -------------------------------------------------------------------
+NEXTTAG := $(shell { git tag --list --merged HEAD --sort=-v:refname; echo v0.0.0; } | grep -E "^v?[0-9]+.[0-9]+.[0-9]+$$" | head -n1 | awk -F . '{ print $$1 "." $$2 "." $$3 + 1 }')
```

Comment:
Similar to what we did in that other repo, I've reworked this a little more into this:
```
release: nexttag  ## Tag and release binaries for different OS on GitHub release
        git tag $(NEXTTAG)
        git push origin $(NEXTTAG)
        goreleaser release --rm-dist

nexttag:
        $(eval NEXTTAG := $(shell $(NEXTTAG_CMD)))

.PHONY: nexttag release

NEXTTAG_CMD := \
        { git tag --list --merged HEAD --sort=-v:refname; echo v0.0.0; } \
        | grep -E "^v?[0-9]+.[0-9]+.[0-9]+$$" \
        | head -n1 \
        | awk -F . '{ print $$1 "." $$2 "." $$3 + 1 }'
```

What do you think?

Reply:
sure thing. fixed.